enhancements/installer/operators-check.md

```diff
@@ -0,0 +1,174 @@
+---
+title: cluster-operator-installation-check
+authors:
+  - @patrickdillon
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - @wking
+  - @LalatenduMohanty
+approvers:
+  - @sdodson
+  - @deads2k
+  - @bparees
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2021-07-14
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - none
+see-also:
+  - "/enhancements/this-other-neat-thing.md"
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Cluster Operator Installation Check
+
+## Summary
+
+This enhancement proposes that the OpenShift Installer should check the status
+of individual cluster operators and use those statuses as a threshold for 
+determining whether an install is a success (exit 0) or failure (exit != 0).
+Specifically, the Installer should check whether cluster operators have stopped
+progressing for a certain amount of time. If the Installer sees that an operator
+is Available=true but fails to enter a stable Progressing=false state, the Installer
+will exit non-zero and log an error.
+
+## Motivation
+
+This enhancement allows the Installer to identify and catch a class of errors
+which are currently going unchecked. Without this enhancement, the Installer can
+(and does) declare clusters successfully installed when the cluster has failing components.
+Adding a check for cluster operator stability will allow normal users and managed services teams
+to more easily identify and troubleshoot issues when they encounter this class of failure. The change
+will also help the Technical Release Team identify this class of problem in CI and triage
+issues to the appropriate operator development team.
+
+### User Stories
```

Comment:
can we think about providing a library with such functionality - so it can be reused in AI as well?

Reply:
What functionality specificall? Scanning operator conditions and ensuring they've all settled for at least 30s?